iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -40,10 +40,37 @@ using iox::runtime::ServiceContainer;
 class ServiceDiscovery_test : public RouDi_GTest
 {
   public:
+    void SetUp() override
+    {
+        searchResultOfFindServiceWithLambda.clear();
+    }
+
+    void TearDown() override
+    {
+    }
+
     iox::runtime::PoshRuntime* runtime{&iox::runtime::PoshRuntime::initRuntime("Runtime")};
     ServiceDiscovery sut;
+
+    static ServiceContainer searchResultOfFindServiceWithLambda;
+    static void lambda(const ServiceContainer& s)
+    {
+        searchResultOfFindServiceWithLambda = s;
+    };
 };
 
+ServiceContainer ServiceDiscovery_test::searchResultOfFindServiceWithLambda;
+
+void compareServiceContainers(const ServiceContainer& lhs, const ServiceContainer& rhs)
```

Comment:
I assume the underlying container lacks the operator (did not check). If so, ok, not the right PR to add it.